src/hmmdwrkr.c

```diff
@@ -191,6 +211,280 @@ worker_process(ESL_GETOPTS *go)
+  int              cnt;
+  int              limit;
+  int              status;
+  int              blk_size;
+  WORKER_INFO     *info       = NULL;
+  ESL_ALPHABET    *abc;
+  ESL_STOPWATCH   *w;
+  ESL_THREADS     *threadObj  = NULL;
+  pthread_mutex_t  inx_mutex;
+  int              current_index;
+  time_t           date;
+  char             timestamp[32];
+
+  int              k;
+  ESL_ALPHABET     *abcDNA = NULL;       /* DNA sequence alphabet                               */
+  ESL_GENCODE      *gcode       = NULL;
+  ESL_GENCODE_WORKSTATE *wrk    = NULL;
+  ESL_SQ           *qsqDNATxt = NULL;    /* DNA query sequence that will be in text mode for printing */
+  ESL_SQ           *qsq      = NULL;		 /* query sequence */
+  P7_TOPHITS       *tophits_accumulator = NULL; /* to hold the top hits information from all 6 frame translations */
+  P7_PIPELINE      *pipelinehits_accumulator = NULL; /* to hold the pipeline hit information from all 6 frame translations */
+  RANGE_LIST       *range_list;  /* (optional) list of ranges searched within the seqdb */
+ 
+
+  if (query->cmd_type == HMMD_CMD_SEARCH) {
+    LOG_FATAL_MSG("ERROR: hmmsearcht option is not implemented", EPERM);
+    return;
+  }
+
+  w = esl_stopwatch_Create();
+  abc = esl_alphabet_Create(eslAMINO);
+  abcDNA = esl_alphabet_Create(eslDNA); 
+
+  if (pthread_mutex_init(&inx_mutex, NULL) != 0) p7_Fail("mutex init failed");
+  ESL_ALLOC(info, sizeof(*info) * (env->ncpus));
+
+  /* Log the current time (at search start) */
+  date = time(NULL);
+  ctime_r(&date, timestamp);
+  printf("\n%s", timestamp);	/* note that ctime_r() leaves \n on end of timestamp  */
+
+  /* initialize thread data */
+  esl_stopwatch_Start(w);
+
+  range_list = NULL;
+        
+  if (esl_opt_IsUsed(query->opts, "--seqdb_ranges")) {
+  ESL_ALLOC(range_list, sizeof(RANGE_LIST));
+     hmmpgmd_GetRanges(range_list, esl_opt_GetString(query->opts, "--seqdb_ranges"));
+  }
+
+    /* Set up the genetic code. Default = NCBI 1, the standard code; allow ORFs to start at any aa
+   */
+  gcode = esl_gencode_Create(abcDNA, abc);
+  esl_gencode_Set(gcode, esl_opt_GetInteger(query->opts, "-c"));  // default = 1, the standard genetic code
+
+  if      (esl_opt_GetBoolean(query->opts, "-m"))   esl_gencode_SetInitiatorOnlyAUG(gcode);
+  else if (! esl_opt_GetBoolean(query->opts, "-M")) esl_gencode_SetInitiatorAny(gcode);      // note this is the default, if neither -m or -M are set
+
+
+  /* Set up the workstate structure, which contains both stateful 
+   * info about our position in <sqfp> and the DNA <sq>, as well as
+   * one-time config info from options
+   */
+  wrk = esl_gencode_WorkstateCreate(query->opts, gcode);
+
+  //if (query->cmd_type == HMMD_CMD_SEARCH) threadObj = esl_threads_Create(&search_thread);
+  //else
+    threadObj = esl_threads_Create(&scan_thread);
+
+
+  if (query->query_type == HMMD_SEQUENCE) {
+    fprintf(stdout, "Search seq %s  [L=%ld]", query->seq->name, (long) query->seq->n);
```

Comment:
Don't use or cast to `long`. If you need a wide int, use int64_t (for example) instead, with explicit integer width. To print an int64_t (etc.) look up use of PRI_d64 (etc.) in printf() documentation.